### PR TITLE
Parametric map generator

### PR DESCRIPTION
**Pull request by valdisz**

### CODE REVIEW

File: neworigins_v2/world.cpp [hunk 1]
```diff
@@ -373,6 +373,44 @@ const char *AGetNameString(int name)
 	return "Error";
 }
 
+// void WriteMapAsJson(ARegionArray* map) {
```
Comment (artyomtrityak) [inline neworigins_v2/world.cpp]: Can be deleted?

File: neworigins_v2/world.cpp [hunk 2]
```diff
@@ -417,7 +461,38 @@ void Game::CreateWorld()
 	SetupNames();
 
 	regions.CreateNexusLevel( 0, nx, ny, "nexus" );
-	regions.CreateSurfaceLevel( 1, xx, yy, 0 );
+	if (generator == 1) {
+		regions.CreateSurfaceLevel( 1, xx, yy, 0 );
+	}
+	else {
+		int continents = -1;
+		while (continents < 16 || continents > 100) {
```
Comment (artyomtrityak) [inline neworigins_v2/world.cpp]: can we calculate this at random based on a map size and water %?

File: neworigins_v2/world.cpp [hunk 3]
```diff
@@ -417,7 +461,38 @@ void Game::CreateWorld()
...
+		while (landMass < 20 || landMass > 100) {
+			Awrite("How % of the surface continents must take? [20..100]");
+			landMass = Agetint();
+		}
+
+		int maxContinentSize = (xx * yy * landMass) / (200 * continents);
+		maxContinentSize = maxContinentSize * 6 / 5;
+		int answer = -1;
+		while (answer < 4 || answer > 5000) {
+			Awrite(AString("How big contients can grow (in regions)? (use 0 for default: ") + maxContinentSize  + ") [4..5000]");
```
Comment (artyomtrityak) [inline neworigins_v2/world.cpp]: misspelled contients -> continents

File: neworigins_v2/world.cpp (hunk 1 above)
Comment (valdisz) [reply neworigins_v2/world.cpp]: I will delete when all changes will be added, i.e. when we will agree that the map look good :)
I helps to debug map a alot

File: neworigins_v2/world.cpp (hunk 2 above)
Comment (valdisz) [reply neworigins_v2/world.cpp]: Yes... but why? This is a tool to set the desired _initial_ granularity of the map. The resulting number of islands still will be random.

**Pull request merged**